src/MemoryRouter.tsx

```diff
@@ -117,8 +119,11 @@ export class MemoryRouter extends BaseRouter {
       this.locale = options.locale;
     }
 
-    this.events.emit("routeChangeComplete");
+    this.events.emit("routeChangeStart", this.asPath, { shallow: false });
+    this.events.emit("routeChangeComplete", this.asPath, { shallow: false });
```

Comment:
Would it make more sense to pass the `options` object?
```suggestion
    this.events.emit("routeChangeComplete", this.asPath, options);
```

Reply:
Going by the docs, I think it's still only passed shallow 🤔 Could extract that obj instead of passing inline too. Also I wasn't sure what machinery happens before calling complete, such as route cancellation. But start simple? 🤷‍♂️

Reply:
You're right, looking through the docs I only see `{ shallow }` as an arg.
I think this would be the best solution, then:
```suggestion
    this.events.emit("routeChangeComplete", this.asPath, { shallow: options?.shallow || false });
```

this would support if someone calls `push(url, as, { shallow: true })` ... 
Not sure if that matters much, but seems like an easy thing to support.